### DISCUSSION TITLE
The flow is never out of view

### DISCUSSION THREAD

**Discussion by FedkoAnastasiia** [xyflow/xyflow#3580] [Q&A] [answered]
I'm trying to make sure that at least one node remains visible when panning or zooming. So that it is not possible to see an empty flow if there is at least one node.

I am adding an images for clarity

For example, this is initial flow: 
![Flow initial](https://github.com/xyflow/xyflow/assets/96294351/f2a1782c-a4a6-4ec9-9db9-c68d65364fe4)

This is unwanted behavior when panning or zooming (none of the nodes are visible in the flow): 
![Flow unvisible](https://github.com/xyflow/xyflow/assets/96294351/b57182b4-b6dc-415c-9760-58db788d3c92)

And this is the desired behavior when panning or zooming (at least one node always remains visible in the flow): 
![Flow desired](https://github.com/xyflow/xyflow/assets/96294351/eab81ef5-f675-4aa1-874a-e0bb603c140e)

Please tell me how this can be implemented. Any working ideas would be appreciated

**Comment by moklick** [comment, marked as answer]
We don't have that feature. In v12 we will make it possible to have a controlled viewport. With that it should be easier to implement a use case like this.

**Reply by FedkoAnastasiia** [reply]
Thank you!
